languages/rust/oso/src/oso.rs

```diff
@@ -151,20 +140,23 @@ impl Oso {
     pub fn register_class(&mut self, class: crate::host::Class) -> crate::Result<()> {
         let name = class.name.clone();
         let name = Symbol(name);
-        let class_name = self.host.lock().unwrap().cache_class(class.clone(), name)?;
-        self.register_constant(&class_name, &class)
+        let class_name = self.host.cache_class(class.clone(), name)?;
+        self.register_constant(&class_name, class)
     }
 
     /// Register a rust type as a Polar constant.
     /// See [`oso::Class`] docs.
-    pub fn register_constant<V: crate::host::ToPolar>(
+    pub fn register_constant<V: crate::host::ToPolar + Send + Sync>(
         &mut self,
         name: &str,
-        value: &V,
+        value: V,
     ) -> crate::Result<()> {
-        let mut host = self.host.lock().unwrap();
         self.inner
-            .register_constant(Symbol(name.to_string()), value.to_polar(&mut host));
+            .register_constant(Symbol(name.to_string()), value.to_polar(&mut self.host));
         Ok(())
     }
 }
+
+// Make sure the `Oso` object is threadsafe
+#[cfg(test)]
+static_assertions::const_assert!(impls::impls!(Oso: Send | Sync));
```

Comment:
> Nice! Why [cfg(test)] though? If we remove that this will just fail to compile if Oso isn't send & sync, which seems right?

Partly so I could make those two import dev-dependencies instead.

> Also, doesn't this check that Oso is Send OR Sync, not Send & Sync?

Good catch, copied from the `Query` one which is doing `!(Send | Sync)`. but failed to properly un-negate the logic

> Why not use assert_impl_all from the static_assertions crate instead of an additional crate (impls)?

Good idea. I found `impls` first and didn't look at what other assertions the other had.